config/v1/types_proxy.go

```diff
@@ -0,0 +1,37 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Proxy holds cluster-wide information on how to configure default proxies for the cluster. The canonical name is `cluster`
+type Proxy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec holds user-settable values for the proxy configuration
+	Spec ProxySpec `json:"spec"`
+}
+
+type ProxySpec struct {
```

Comment:
Please remove `ProxyConfig` in `types_build.go` and replace it with this once #155 merges.

Reply:
> Please remove `ProxyConfig` in `types_build.go` and replace it with this once #155 merges.

As you wish.  Remember that you cannot update docs here to match specific build needs and that when you create your own struct, you need to keep serialization compatibility.